ui/src/app/layout/applications/applications.component.ts

```diff
@@ -262,7 +262,8 @@ export class ApplicationsComponent implements OnInit {
             expiryWarningHtml: of('')
         };
         this.certificateService.getApplicationCertificatesPromise(app.id).then(certs => {
-            this.certificateDlgData.existingCertificate = certs.find(cert => cert.environmentId === env.id) || null;
+            this.certificateDlgData.existingCertificate =
+                certs['authentications'].hasOwnProperty(env.id) ? certs['authentications'][env.id]['authentication'] : null;
```

Comment:
For compiler safety, I'd appreciate using direct properties here. Maybe we have to extend the type declaration... (This applies to all the other `['authentications']` etc. references as well)